cupy/functional/piecewise.py

```diff
@@ -0,0 +1,76 @@
+import cupy
+
+from cupy import core
+from cupy.core.core import ndarray
+from cupy.core._reduction import ReductionKernel
+
+_piecewise_krnl = ReductionKernel(
+    'S x1, T x2',
+    'U y',
+    'x1 ? x2 : NULL',
+    'b == NULL? a : b',
+    'y = a',
+    'NULL',
+    'piecewise'
+)
```

Comment:
`ReductionKernel` performs well for a large size reduction axis.
However, int the case of this routine, the parameter `condlist` is given as a list of ndarrays or scalars -- its size is equal to the reduction size -- and [concatenating them](https://github.com/cupy/cupy/pull/3329/files#diff-26cc69efdac81a9dee1d8f0ebc5d0356R54) cannot be parallelized. So this function cannot work with good performance for a large `condlist` size even if we implement it with `ReductionKernel`.
How about using `ElementwiseKernel` in a loop simply? If we do so, we don't have to concatenate `condlist`, and we don't need to allocate additional memory space.